subworkflows/nf-core/fasta_ltrharvest_ltrfinder_ltrretriever_lai/main.nf

```diff
@@ -0,0 +1,291 @@
+                                        def id = gff.name.split('.restored.ids.gff3')[0]
+
+                                        [ id, gff ]
+                                    }
+                                    | join(
+                                        ch_gff_tsv_branch.change
+                                        | map { meta, gff, tsv -> [ meta.id, meta ] }
+                                    )
+                                    | map { id, gff, meta -> [ meta, gff ] }
+
+
+    ch_restored_gff                 = ch_gff_tsv_branch.nochange
+                                    | map { meta, gff, tsv -> [ meta, gff ] }
+                                    | mix(ch_collected_gff)
+
+    emit:
+    ltrlib                          = ch_ltrlib         // channel: [ val(meta), fasta ]
+    annotation_gff                  = ch_restored_gff   // channel: [ val(meta), gff ]
+    lai_log                         = ch_lai_log        // channel: [ val(meta), log ]
+    lai_out                         = ch_lai_out        // channel: [ val(meta), out ]
+    versions                        = ch_versions       // channel: [ versions.yml ]
+}
+
+
+def map_monoploid_seqs_to_new_ids(id, short_ids_tsv, monoploid_seqs) {
+
+    def short_ids_head = short_ids_tsv.text.tokenize('\n')[0]
+
+    if (short_ids_head == "IDs have acceptable length and character. No change required.") {
+        return [ "${id}.mapped.monoploid.seqs.txt" ] + monoploid_seqs.text.tokenize('\n')
+    }
+
+    def orig_to_new_ids = [:]
+    short_ids_tsv.text.eachLine { line ->
+        def (original_id, renamed_id) = line.tokenize('\t')
+        orig_to_new_ids[original_id] = renamed_id
+    }
+
+    def mapped_ids = []
+    monoploid_seqs.text.eachLine { original_id ->
+        if (!orig_to_new_ids[original_id]) {
+            error "Faild to find $original_id in ${short_ids_tsv}" +
+            "\nThe short_ids_tsv file is malformed!"
+        }
+
+        mapped_ids.add(orig_to_new_ids[original_id])
+    }
+
+    return [ "${id}.mapped.monoploid.seqs.txt" ] + mapped_ids
+}
+
+
+def restore_gff_ids(id, file_gff, short_ids_tsv) {
+    def new_to_orig_ids = [:]
+
+    short_ids_tsv.text.eachLine { line ->
+        def columns = line.tokenize('\t')
+
+        def orig_id = columns[0]
+        def new_id  = columns[1]
+        new_to_orig_ids[new_id] = orig_id
+    }
+
+    def restored_gff = []
+
+    file_gff.text.eachLine { line ->
+        if ( line.startsWith('##') ) {
+            restored_gff.add(line)
+            return
+        }
+
+        def columns = line.tokenize('\t')
+        def new_id = columns[0]
+        def orig_id = new_to_orig_ids[new_id]
+
+        restored_gff.add ( ( [ orig_id ] + columns[1..-1] ).join('\t') )
+    }
+
+    [ "${id}.restored.ids.gff3", restored_gff.join('\n') ]
+
+}
+
+def do_ids_need_to_change(ids_descs) {
+
+    if ( ids_descs.collect { it.id.size() > 13      }.any() ) return true
```

Comment:
You should be able to supply the test to `any` and then just chain the any statements in a single return rather than having `return true`/`return false`.